### PR TITLE
Add advanced code example to documentation

### PR DESCRIPTION
**Pull request by Alex1607**
**What this PR solves / how to test:**
In my opinion, removing the code example and just using a bare-bones file leaves a lot of new starters in the dark. Therefore, I added the old code as an advanced example to the documentation.

**Associated docs issue(s)/PR(s):**
- #2936

**Author has included the following, where applicable:**

- [ ] Tests
- [ ] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer has performed the following, where applicable:**

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
Please run Prettier 
`npm run fix`

File: templates/experimental/worker-rust/README.md [hunk 1]
```diff
@@ -36,6 +36,62 @@ $ npm run deploy
 
 Read the latest `worker` crate documentation here: https://docs.rs/worker
 
+## Advanced Example
+
+Given that this template only contains the most basic setup, you might want to use this more advanced example. It demonstrates how you can create multiple routes, log requests, get field data from a form and more.
```
Comment (JacobMGEvans) [inline templates/experimental/worker-rust/README.md:41]:
```suggestion
As this template comprises only the essential setup, we recommend considering our advanced example to leverage its additional functionalities. The advanced example showcases the creation of multiple routes, logging of requests, retrieval of field data from a form, and other features that may prove useful to your project.
```

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
The messaging needs some rewording and the formatter needs to be ran on the file. If the formatter makes the snippet look off we can add that to the prettier/linter ignore

**Comment by Alex1607** [comment]
Checks are failing because a test timed out: `→ Test timed out in 30000ms.`
I don't think it's related to my changes.

**Comment by nouzu** [comment]
I suggest we use example from the workers-rs repo, it's compact and does the job.

```rust
use worker::*;

#[event(fetch)]
pub async fn main(req: Request, env: Env, _ctx: worker::Context) -> Result<Response> {
    console_log!(
        "{} {}, located at: {:?}, within: {}",
        req.method().to_string(),
        req.path(),
        req.cf().coordinates().unwrap_or_default(),
        req.cf().region().unwrap_or("unknown region".into())
    );

    if !matches!(req.method(), Method::Post) {
        return Response::error("Method Not Allowed", 405);
    }

    if let Some(file) = req.form_data().await?.get("file") {
        return match file {
            FormEntry::File(buf) => {
                Response::ok(&format!("size = {}", buf.bytes().await?.len()))
            }
            _ => Response::error("`file` part of POST form must be a file", 400),
        };
    }

    Response::error("Bad Request", 400)
}
```

**Review by JacobMGEvans — APPROVED** [review]

**Comment by jspspike** [comment]
@Alex1607 Could this link to `workers-rs` README? Since this is just a copy of that I feel like that would make sense in the case the reader wants to read more or any changes are made to this example based on changes in `workers-rs`. Honestly I feel like it would make the most sense to remove the snippet entirely and just say you could go to this link for more info but if you feel like it makes sense to include it that's fine with me

**Comment by Alex1607** [comment]
@jspspike @lrapoport-cf sorry for the delay, I just adjusted the readme

**Review by jspspike — APPROVED** [review]
Looks good to me

**Comment by Alex1607** [comment]
How is the merging done? Do I still have to do something, or will someone take over from now?

**Comment by lrapoport-cf** [comment]
thanks @Alex1607 , we've merged the change 👍

**Pull request merged**